### DISCUSSION TITLE
Setting up workers

### DISCUSSION THREAD

**Discussion by vtosevski** [photoprism/photoprism#1921] [Q&A] [answered]
Dear photoprism team,

Thanks for maintaining such a cool project!
I have just recently installed the photoprism in a VM for testing purposes and was wondering how does one set up the number of workers? The host OS is Windows server 2016 running with 16GB RAM and on 4 cores/8 threads. Guest OS is Ubuntu server with 2 cores and 4 threads assigned to it.

Upon photoprism installation I don't see any line in docker-compose.yml that pertains to workers.
Running `docker-compose exec photoprism photoprism config` returns 1 as number of workers.

I have then added manually a line in the docker-compose.yml that specifies 4 workers (per instructions [here](https://docs.photoprism.app/getting-started/config-options/)) and restarted the application. Checking the config again as above still returns 1. Is this expected? Am I trying to set up workers in the right way at all?

Thanks,
Vinko

**Comment by vtosevski** [comment, marked as answer]
For future reference, I was able to solve my issues over at https://gitter.im/browseyourlife/community#

In short:

> Michael Mayer
> Could this be true, somehow? https://github.com/photoprism/photoprism/blob/develop/internal/config/config.go#L483

Thanks all for the feedback! I don't have a habit of checking the source code first but thanks to
Michael Mayer link I could learn that photoprism assignes one worker whenever there's less RAM than "minimum amount" (default of 5 GB). My VM was set with dynamic memory allocation, starting from 2 GB, so it always resulted in one worker, no matter what the setting in `docker-compose.yml` was. Setting up RAM allocation to > 5GB indeed resulted in multiple workers being available. Kudos and thanks!

**Reply by lastzero** [reply]
Looking at the commit comment, maybe 5 GB was a typo on my side... it says "if < 3 GB of memory" 🧐

**Reply by lastzero** [reply]
Reduced starting with the next development preview. Probably it was increased to 5 because too many users did not have swap configured and we were flooded with "bug reports".
